openshift/recovery-controller/src/main/java/me/snowdrop/boot/narayana/openshift/recovery/NarayanaRecoveryTerminationControllerAutoConfiguration.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2018 Red Hat, Inc, and individual contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package me.snowdrop.boot.narayana.openshift.recovery;
+
+import java.util.List;
+
+import org.springframework.boot.autoconfigure.AutoConfigureAfter;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.transaction.jta.NarayanaJtaConfiguration;
```

Comment:
This is the old configuration. I think to make sure it works with SB v1 and v2 you could add @AutoConfigureAfter({"me.snowdrop.boot.narayana.SpringBoot1NarayanaConfiguration", "me.snowdrop.boot.narayana.SpringBoot2NarayanaConfiguration"}). Only one of them should be in the class path, but with the string rather than the class it should work.